MandatoryTask/MandatoryTask/Candy.java

```diff
@@ -0,0 +1,52 @@
+package MandatoryTask;
+
+import java.util.Comparator;
+
+public abstract class Candy implements Comparable<Candy> {
+    private String name;
+    private int weightOfOneCandyGram;
+    private int calorificPerGram ;
+
+    public Candy(String name, int weightOfOneCandyGram, int calorificPerGram) {
+        this.name = name;
+        this.weightOfOneCandyGram = weightOfOneCandyGram;
+        this.calorificPerGram = calorificPerGram;
+    }
```

Comment:
Валідацію параметрів краще перенести сюда, так як ми хочемо щоб не валідні цукерки навіть не створювалися 
```suggestion
    public Candy(String name, int weightOfOneCandyGram, int calorificPerGram) {
        if (name == null || name.length() == 0) {
            throw new IllegalArgumentException("Candy name cannot be null or empty");
        }
        if (weightOfOneCandyGram <= 0 ) {
            throw new IllegalArgumentException("Candy weight cannot be 0 or less");
        }
        if (calorificPerGram <= 0 ) {
            throw new IllegalArgumentException("Candy calorificPerGram cannot be 0 or less");
        }
        this.name = name;
        this.weightOfOneCandyGram = weightOfOneCandyGram;
        this.calorificPerGram = calorificPerGram;
    }
```